snapcraft/internal/errors.py

```diff
@@ -708,3 +708,10 @@ def get_resolution(self) -> str:
             "the data source of the long data value. In some cases, the "
             "filesystem being used will limit the allowable size."
         )
+
+
+class CorruptedElfFile(SnapcraftError):
```

Comment:
Can you use SnapcraftException instead of SnapcraftError?  We're trying to migrate away from SnapcraftError.

Reply:
Of course, thanks for the heads up: done. Note that I added a mitigation that, in the current codebase, will never been seen. Still seems sensible, though.